### ISSUE TITLE
REST API: add endpoints for calculation input and output files

### ISSUE THREAD

**Issue by waychal** [aiidateam/aiida-core#957]
For calculation:
- add endpoint to return list of all input and output files
- add endpoint to download input and output file

**Comment by waychal** [comment]
add a test where you try to get a file that exists, and another where you try to get a file that does not exist, and a third one where you try to get a file out of the folder with many '..' and check that it complains.

**Comment by dev-zero** [comment]
I have something similar implemented in another project. Inside one of my calculation resources I return the following:

```json
  "infiles": [
    {
      "_links": {
        "collection": "http://server/base/api/v2/artifacts", 
        "download": "http://server/base/api/v2/artifacts/b560c427-8b72-4264-a896-f0f36f5b549f/download", 
        "self": "http://server/base/api/v2/artifacts/b560c427-8b72-4264-a896-f0f36f5b549f", 
        "view": "http://server/base/api/v2/artifacts/b560c427-8b72-4264-a896-f0f36f5b549f/view"
      }, 
      "id": "b560c427-8b72-4264-a896-f0f36f5b549f", 
      "metadata": {
        "compressed": null
      }, 
      "name": "BASIS_SETS"
    }
  ],
  "outfiles": [
  ]
```

Now, all of my REST records contain this `_links` dict with a link to the original resource collection and to themselves. Artifacts (generalized files) got two additional entries there: `download` and `view`. I use those two different endpoints (sub-resources) to control the `Content-Disposition` on the server side to be able to directly return the URL to the user of a web-interface to either view the file in a new tab or then download the file. The `self` resource returns a more complete json to what is already inside my calculation object with additional metadata.

**Comment by waychal** [comment]
This issue is already fixed. The only part left is to add test cases.

**Comment by sphuber** [comment]
Fixed in PR #3429
